### ISSUE TITLE
Add support for `TyperArgument` arguments

### ISSUE THREAD

**Issue by taranlu-houzz** [ewels/rich-click#59]
It seems like `typer.Argument` params are being grouped in with the other options. This happens whether or not `SHOW_ARGUMENTS` is enabled.

![image](https://user-images.githubusercontent.com/52425255/164339457-1377ce15-e2f2-4fbe-8051-bfa10de624ef.png)

**Comment by ewels** [comment]
Thanks for reporting this! I was checking for `click.core.Argument` but Typer has it's own type, `typer.core.Argument` as you said.. So now the code should detect either of these types and skip unless `SHOW_ARGUMENTS` is specified.

Let me know how you get on with it!

Cheers,

Phil

**Comment by ewels** [comment]
Ah that's no good, thanks for letting me know. I've moved your comment to a new issue (#70) to track. I'll fix + release ASAP.

**Comment by taranlu-houzz** [comment]
@ewels Hey, thanks for the quick fix. The error is gone now, but I still am having the initial issue (using the latest version of `rich-click` that you just released). Even though I have `SHOW_ARGUMENTS` set, it still groups them under options as shown in the initial screenshot I posted.

I have also tried explicitly setting `GROUP_ARGUMENTS_OPTIONS = False`, but the did not help.

**Comment by ewels** [comment]
Ah really? It should now hide the arguments by default.. 🤔 (If you set `SHOW_ARGUMENTS` to `True` it'll go back to the same behaviour you saw above).

It works for me with [`examples/typer/02_subcommands.py`](examples/typer/02_subcommands.py) (note `NAME` is a positional argument and doesn't show in the table):

```console
$ python examples/typer/02_subcommands.py download --help
Debug mode is off

 Usage: 02_subcommands.py download [OPTIONS] NAME

 Pretend to download some files from somewhere. Multi-line help strings are unwrapped until you use a double newline.
 Only the first paragraph is used in group help texts. Don't forget you can opt-in to rich and markdown formatting!
 Click escape markers should still work.
   * So you
   * Can keep
   * Your newlines

 And this is a paragraph that will be rewrapped again.

╭─ Options ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│ --all/--no-all      Get everything [default: no-all]                                                                                       │
│ --help              Show this message and exit.                                                                                            │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯

```

Can you add an example script to this issue and what you expect it to show?

It's entirely possible that `GROUP_ARGUMENTS_OPTIONS` doesn't work with Typer arguments. They're a different type to Click and I don't remember messing with that code so I probably missed that 🤔

**Comment by ewels** [comment]
Yup, there are a few instances of `click.core.Argument` which need `TyperArgument` support adding to them..

**Comment by taranlu-houzz** [comment]
Hey, sorry! Unfortunately, I don't have time to cleanup an example right now (will try to later if still an issue). It sounds like you have managed to identify where the problem is though, right?

**Comment by ewels** [comment]
> It sounds like you have managed to identify where the problem is though, right?

Potentially, but I'm not sure it's the same - I was able to replicate your original issue to begin with but not since 0956afe15feddd31397f64f5073e04d290d4b368

I'll try to find some time to play with the `TyperArgument` stuff a bit more if I can, it will hopefully solve your issues. It's just a little risky without a test case as if I can't replicate your issue I don't know if I've fixed it or not 😉

**Comment by taranlu-houzz** [comment]
Makes sense. I will try to get a cut down example to you when I have a chance.

**Comment by AlirezaTheH** [comment]
Any update on this?

**Comment by ewels** [comment]
Nope, still need a minimal example to replicate the problem.

**Comment by zmoon** [comment]
How about this?

```python
from rich_click import typer

typer.rich_click.SHOW_ARGUMENTS = True


def cli(
    name: str = typer.Argument("World"),
    say_hi: bool = typer.Option(True),
):
    if say_hi:
        print(f"Hello, {name}!")


if __name__ == "__main__":
    typer.run(cli)
```

The argument shows with `SHOW_ARGUMENTS = True` set, but in the Options section.

![image](https://user-images.githubusercontent.com/15079414/177662342-55f9cafd-3412-4f60-9b80-dcdf649a33d2.png)

**Comment by AlirezaTheH** [comment]
@ewels If you need any help on this just say the word.

**Comment by ewels** [comment]
I've just started my summer holiday so I'll be slow to tackle this now. So yeah if you fancy taking a crack at it that would be fab 👍🏻

**Comment by AlirezaTheH** [comment]
OK, I'll give it a try.

**Comment by taranlu-houzz** [comment]
@AlirezaTheH @ewels @zmoon Hey, sorry I never got back on this. Thanks for taking things forward!

**Comment by ewels** [comment]
Fixed in #82
